src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3098,29 +3136,48 @@ ss::future<> tx_gateway_frontend::expire_old_txs(ss::shared_ptr<tm_stm> stm) {
 
 ss::future<> tx_gateway_frontend::expire_old_tx(
   ss::shared_ptr<tm_stm> stm, kafka::transactional_id tx_id) {
-    return with(stm, tx_id, "expire_old_tx", [this, stm, tx_id]() {
-        return do_expire_old_tx(
-          stm, tx_id, config::shard_local_cfg().create_topic_timeout_ms());
-    });
-}
+    auto units = co_await stm->lock_tx(tx_id, "expire_old_tx");
 
-ss::future<> tx_gateway_frontend::do_expire_old_tx(
-  ss::shared_ptr<tm_stm> stm,
-  kafka::transactional_id tx_id,
-  model::timeout_clock::duration timeout) {
     auto term_opt = co_await stm->sync();
     if (!term_opt.has_value()) {
+        if (term_opt.error() == tm_stm::op_status::not_leader) {
+            vlog(
+              txlog.trace,
+              "this node isn't a leader for tx.id={} coordinator",
+              tx_id);
+        }
+        vlog(
+          txlog.warn,
+          "got error {} on syncing state machine (loading tx.id={})",
+          term_opt.error(),
+          tx_id);
         co_return;
     }
+
     auto term = term_opt.value();
+
+    co_await do_expire_old_tx(
```

Comment:
Why are we ignoring an error here ?

Reply:
`do_expire_old_tx` is triggered by a timer so there is nowhere to propagate the error and any transient error will be avoided by next iteration